### ISSUE TITLE
Add support of @type

### ISSUE THREAD

**Issue by robertschubert** [decentralized-identity/jsonld-common-java#13]
I am using the `verifiable-credentials-java` module.

When executing `credential.getCredentialSubject().getTypes()` the method returns the values of the field with the name `type` in the `credentialSubject`.

Looking into the code obviously the type is returned in `JsonLDObject`:

```
public final String getType() {
        return JsonLDUtils.jsonLdGetString(this.getJsonObject(), "type");
    }
```

I have a json object defined like this:

```
{
    "@context": {
      "xsd": "http://www.w3.org/2001/XMLSchema#",
    },
    "@id": "my-id",
    "@type": "my-type",
```

When I understood it right, `@type` is a keyword in JSON-LD used to set the type of an object, while `type` is a term that is defined by the `@context` and its meaning it can vary based on the context.

In the current implementation I cannot get the `@type` since only `type` is processed.

Can you add support for `@type` in the `JsonLDObject`?

**Comment by robertschubert** [comment]
BTW: workaround for me as user of that library is to call `credential.getCredentialSubject().getJsonObject()` and then get `@type` from the map.

**Comment by peacekeeper** [comment]
> When I understood it right, `@type` is a keyword in JSON-LD used to set the type of an object, while `type` is a term that is defined by the `@context` and its meaning it can vary based on the context.

@robertschubert You are correct about this from a pure JSON-LD perspective. However, the W3C VCDM specification is clear that it uses `type`, not `@type`. See [here](https://www.w3.org/TR/vc-data-model/#types) and [here](https://www.w3.org/TR/vc-data-model-2.0/#types).

Therefore I think our implementation should not support a dedicated call to get `@type`, and if you really need that, you coud use the workaround you mention.

Do you agree, or do you see that differently?

**Comment by robertschubert** [comment]
Hi @peacekeeper 
thanks for your answer. I can live with the workaround. 

But: When going through both links you provided there is a note in both (just ctrl - f "@type"):

> NOTE: The Verifiable Credentials Data Model is based on JSON-LD
> The [type](https://www.w3.org/TR/vc-data-model-2.0/#dfn-type) system for the Verifiable Credentials Data Model is the same as for [[JSON-LD11](https://www.w3.org/TR/vc-data-model-2.0/#bib-json-ld11)] and is detailed in [Section 3.5: Specifying the Type](https://www.w3.org/TR/json-ld/#specifying-the-type) and [Section 9: JSON-LD Grammar](https://www.w3.org/TR/json-ld/#json-ld-grammar). When using a JSON-LD context (see Section [5.2 Extensibility](https://www.w3.org/TR/vc-data-model-2.0/#extensibility)), this specification aliases the @type keyword to type to make the JSON-LD documents more easily understood. While application developers and document authors do not need to understand the specifics of the JSON-LD type system, implementers of this specification who want to support interoperable extensibility, do.
> 

If I understand it correctly an implementation should alias the `@type` to `type`. So I guess it should be supported from a specification point of view.

**Comment by peacekeeper** [comment]
This should be fixed by https://github.com/decentralized-identity/jsonld-common-java/commit/aa8eba0d5402ac3ad0e0bae51165bc6a7783f115

**Comment by robertschubert** [comment]
Hi @peacekeeper 
cool, looks good! Thanks for hinting this!
Once the change is released we can close this issue

**Comment by peacekeeper** [comment]
@robertschubert Should be fixed and released now in version 1.11.0
 
 https://github.com/decentralized-identity/jsonld-common-java/releases/tag/1.11.0
